cpp/src/arrow/stl.h

```diff
@@ -55,13 +67,55 @@ ARROW_STL_CONVERSION(uint32_t, UInt32Type)
 ARROW_STL_CONVERSION(uint64_t, UInt64Type)
 ARROW_STL_CONVERSION(float, FloatType)
 ARROW_STL_CONVERSION(double, DoubleType)
-ARROW_STL_CONVERSION(std::string, StringType)
+
+template <>
+struct ConversionTraits<std::string> {
+  using ArrowType = StringType;
+  static std::shared_ptr<DataType> arrow_type() { return std::make_shared<StringType>(); }
+  static Status AppendRow(StringBuilder& builder, const std::string& cell) {
+    return builder.Append(cell);
+  }
+  static std::string GetEntry(const StringArray& array, size_t j) {
+    return array.GetString(j);
```

Comment:
Perhaps you can use `GetView` here. It also exists on numeric arrays, so you wouldn't need the `std::string` specialization.

Reply:
I still would need a specialization here as in the end I still want to have a `std::string` instance in the resulting vector.